pkg/executor/guest.go

```diff
@@ -48,9 +49,22 @@ func IsFileExists(
 	}
 	execErr := execCommand(cr, podName, command, ioStreams)
 	if execErr != nil {
-		return false
+		shared.LogErrorf(
+			cr,
+			"command{%s} IsFileExists: %v",
+			command,
+			execErr,
+		)
+
+		// If the command fails with the error "command terminated with exit code 1",
+		// this means the file existence check completed successfully, but the file does not exist.
+		// Otherwise the command failed for some other reason.
+		if strings.Compare(strings.TrimRight(execErr.Error(), "\r\n"), "command terminated with exit code 1") == 0 {
```

Comment:
I went looking for where this message is generated, just to see how brittle it would be to depend on this particular format. From poking around in there, I think we don't need to string munge to get our answer here. E.g. we could check to see if `execErr.(type) == exec.CodeExitError` (where `exec` is the import of "k8s.io/client-go/util/exec").

Further, if we know the type is CodeExitError then we could also call `execErr.ExitStatus()` to [get an integer exit status directly](https://godoc.org/k8s.io/client-go/util/exec), although we probably don't care in this particular case.

Reply:
Probably we can/should remove the preceding LogErrorf as well. It's going to be fairly normal/usual for file existence checks to fail (especially once PR #62 goes in).